File: packages/dart/lib/src/word/rand_toy.dart
import 'dart:math';

import 'package:randino/src/types.dart';
import 'package:randino/src/word/rand_word.dart';

/// Toys and games, the things and the play a childhood is made of.
///
/// [randWord] with the theme already chosen; every other parameter is the same.
/// For the detail form, pass [WordTheme.toy] to `randWordDetails` — Dart has
/// no overloads, and a second function per theme would be a second wall of them.
///
/// ```dart
/// randToy(language: WordLanguage.ko, count: 3); // [팽이, 연, 딱지]
/// randToy(language: WordLanguage.en, count: 3); // [Kite, Yoyo, Domino]
/// ```
List<String> randToy({
  WordLanguage? language,
  int count = 1,
  RandRealism realism = RandRealism.real,
  RandVocabulary vocabulary = RandVocabulary.full,
  int? minLength,
  int? maxLength,
  String? startsWith,
  bool unique = false,

  /// Where the randomness comes from: `Random.secure()` for a value nobody may
  /// predict, `Random(42)` for one that has to come out the same every run.
  Random? random,
}) => randWord(
  language: language,
  theme: WordTheme.toy,
  count: count,
  realism: realism,
  vocabulary: vocabulary,
  minLength: minLength,
  maxLength: maxLength,
  startsWith: startsWith,
  unique: unique,
  random: random,
);

File: packages/javascript/lib/word/randToy.ts
import type { RandThemedWordOptions, WordDetail } from '../_types/global.js';
import { themedWord } from './wordGenerator.js';

/**
 * Toys and games, the things and the play a childhood is made of.
 *
 * `randWord({ theme: 'toy' })` with the theme already chosen; every other
 * option is the same.
 *
 * @example
 * randToy({ language: 'ko', count: 3 }); // ['팽이', '연', '딱지']
 * randToy({ language: 'en', count: 3 }); // ['Kite', 'Yoyo', 'Domino']
 */
export function randToy(options?: RandThemedWordOptions & { output?: 'value' }): string[];
/**
 * Toy words with the language and theme each one came from.
 *
 * @example
 * randToy({ language: 'ko', output: 'detail' });
 * // [{ word: '팽이', language: 'ko', theme: 'toy' }]
 */
export function randToy(options: RandThemedWordOptions & { output: 'detail' }): WordDetail[];
export function randToy(options: RandThemedWordOptions = {}): string[] | WordDetail[] {
	return themedWord('toy', options);
}
